### DISCUSSION TITLE
Firmware OpenDTU

### DISCUSSION THREAD

**Discussion by webwerb** [tbnobody/OpenDTU#252] [General]
Moin,

hab seit ein paar Wochen den Hoymiles 1500 in einer provisorischen Auftstellung im Test laufen bevor ich alles engültig montiere. Dazu OpenDTU welches ich über die Diskussion im microcontroller.net gefunden habe.

Läuft bisher super und es zeigt wieder einmal wozu die community in der Lage ist; vielen Danke an alle Beteiligten dafür!

Nächster Schritt nach der Montage ist die Integration in meine homematic-Umgebung; vermutlich über über Homeastant(?).

Als Windows User mit mäßgen Kenntnissen zu den ESP´s (und auch Linux - es reicht gerade mal für einen Raspi mit pihole ...) habe ich nach dem download für die Firmware des OpenDTU gesucht und bisher nicht gefunden.

Hat da jemand einen Tip/link für mich?

Danke!

**Comment by dersch81** [comment]
Es gibt keine fertige FW zum Download. Du muss das mit VScode selbst bauen. Aber das ist überhaupt kein Problem und in der Readme Schrittweise beschrieben.

**Reply by webwerb** [reply]
Wie man unten lesen kann, stimmt das nicht.
Man kann auf den beschriebenen Wegen sehr wohl aktuelle Firmware bekommen und OTA installieren.
Nur damit sich hier niemand beim Lesen ins Bockshorn jagen lässt.

**Comment by webwerb** [comment]
Seltsam, unter "System Info" "Firmware Update" wird mir angezeigt "New Version available. Show Changes"

**Comment by dersch81** [comment]
Das ist korrekt. Es bezieht sich auf die letzten commits im master branch. Ich sehe aber grade in der Readme ist auch eine Beschreibung zu pre-complied binarys. Habe sie aber auf die schnelle nicht gefunden.

**Comment by webwerb** [comment]
Verstehe ich leider nicht, da ich in diesem Bereich nicht zuhause bin.
Ich kenne es von anderen Vorhaben so, das ich eine neue Firmware herunterladen und OTA installieren kann.
Ich kann mich hier leider nicht weiter einarbeiten, da mir dafür die Zeit fehlt. Erstmal muss die PV aufs Dach und angeschlossen werden.
Hab ich eben Pech gehabt und werde mit der bisherigen Firmware leben. Funktioniert ja.

**Comment by ghost** [comment]
Unter GitHub / OpenDTU / Actions gibt es „Workflows“. Bei vielen davon, aber nicht bei allen (warum auch immer) wird nach Anklicken unten auf der Seite  unter „Artifacts“ „opendtu-generic“ sichtbar. Nach Anmelden bei GitHub kannst du eine zip-Datei herunterladen. Nach Entpacken kannst du mit „opendtu-generic.bin“ die Firmware updaten.

**Reply by CONSULitAS** [reply]
> Bei vielen davon, aber nicht bei allen (warum auch immer) wird nach Anklicken unten auf der Seite unter „Artifacts“ „opendtu-generic“ sichtbar.

Das liegt an der Art "Action" die gelaufen ist. 

Standardmäßig zeit der Klick GitHub -> OpenDTU -> Actions "All Workflows" an.

Mit Klick auf "OpenDTU build" links (https://github.com/tbnobody/OpenDTU/actions/workflows/build.yml) werden nur die Actions mit Build-Prozess gefiltert. In denen ist dann immer ein `.bin`, wenn der Status "success" (als grüner Haken) ist. Das kann man dann noch über die Filter einschränken (https://github.com/tbnobody/OpenDTU/actions/workflows/build.yml?query=is%3Asuccess). Ist aber eigentlich unnötig.

**Reply by ghost** [reply]
Danke für die Erklärung - klappt mit Filter. Ich hatte bisher selbst gefiltert und jeweils die Actions mit Build-Process herausgesucht.

**Comment by webwerb** [comment]
Danke! Dann probiere ich das mal.

**Comment by FrodoVDR** [comment]
Der letzte commit liegt da: https://github.com/tbnobody/OpenDTU/actions/runs/3292595274

**Comment by webwerb** [comment]
Prima, das hat funktioniert.
Danke!
